# Oppgave sekvensiell konkurranse (Stackelberg)

For å møte fremtidige vekstmuligheter i markedet har Hurtigruten ASA valgt å spisse de ulike delene av konsernet. Selskapet har opprettet enhetene Hurtigruten Norge, som skal drive ordinær rutetrafikk langs norskekysten, og Hurtigruten Expeditions, som skal drive den internasjonale ekspedisjonscruise-virksomheten.

I markedet for ekspedisjonscruise vil Hurtigruten Expeditions være en ledende aktør, hvor de vil ha mulighet til å gjøre sine valg av kapasitet før en eventuell konkurrent kommer inn i markedet.

I markedet for ekspedisjonscruise vil total invers etterspørsel være gitt ved:

$$
P = 5000 - \frac{1}{10} \cdot (q_E + q_N)
$$

hvor $q_E$ er antall passasjerer som vil reise med Hurtigruten Expeditions, $q_N$ er antall passasjerer som vil reise med nykommeren og $P$ er pris på cruise. På ekspedisjonscruise vil marginalkostnadene per passasjer være noe høyere siden det her er lagt opp til flere utflukter under seilasen. Hurtigruten Expeditions vil ha konstante marginalkostnader på 1500 per passasjer, mens nykommeren er noe mindre effektiv i sin drift og vil ha marginalkostnader på 1800 per
passasjer. 

Faste kostnader for de nye cruiseskipene er på 3 millioner.

a) Hva blir optimal tilpasning i dette markedet når Hurtigruten Expeditions kan gjøre sitt valg av kapasitet før nykommeren velger sitt kvantum.

Vil det være en fordel for Hurtigruten Expeditions å ha mulighet til å gjøre valg av kvantum før konkurrenten?

In [1]:
import numpy as np
import sympy as sp

In [2]:
# Definerer symbol
P,Q,q_1,q_2,a,b,c,FK = sp.symbols('P Q q_1 q_2 a b c FK')

# Q = q_1 + q_2
Q = q_1 + q_2

a = 5000
b = 1/10
c_1 = 1500
c_2 = 1800
Faste_kostnader = 3000000

# Definerer etterspørselsfunksjonen
Etterspørsel = sp.Eq(Q, (a-P)/b)

# Løser for P
Invers_etterspørsel = sp.solve(Etterspørsel, P)[0]

# Viser løsningene
display(Etterspørsel, Invers_etterspørsel)

Eq(q_1 + q_2, 50000.0 - 10.0*P)

-0.1*q_1 - 0.1*q_2 + 5000.0

In [3]:
# Multipliserer prisen med kvantumet for å finne totalinntekten
Totalinntekt = Invers_etterspørsel*q_1
Totalinntekt2 = Invers_etterspørsel*q_2

# Viser totalinntekten
display(Totalinntekt, Totalinntekt2)

q_1*(-0.1*q_1 - 0.1*q_2 + 5000.0)

q_2*(-0.1*q_1 - 0.1*q_2 + 5000.0)

In [4]:
# Deriverer totalinntekten for å finne marginalinntekten
Marginalinntekt_q2 = sp.diff(Totalinntekt2, q_2)

# Viser marginalinntekten
display(Marginalinntekt_q2)

-0.1*q_1 - 0.2*q_2 + 5000.0

In [5]:
# Trinn 1

# Setter marginalinntekten lik kostnaden
likevekt_q2 = sp.Eq(Marginalinntekt_q2, c_2)

display(sp.simplify(likevekt_q2))

Eq(0.1*q_1 + 0.2*q_2, 3200.0)

In [6]:
# Løser for q2 for å finne reaksjonsfunksjon til bedrift 2
q2 = sp.solve(likevekt_q2, q_2)[0]

display(q2)

16000.0 - 0.5*q_1

In [7]:
# Trinn 2

# Vi setter så reaksjonsfunksjonen til bedrift 2 inn i bedrift 1 sin Totalinntekt og deriverer med hensyn på q1
profitt_1_q = sp.diff(Totalinntekt.subs(q_2, q2), q_1)

display(profitt_1_q)

3400.0 - 0.1*q_1

In [8]:
# Finner optimalt kvantum for lederbedrift ved å sette profitten lik kostnaden og løse for q1
q1_sol = sp.solve(sp.Eq(profitt_1_q, c_1), q_1)[0]

display(q1_sol)

19000.0000000000

In [9]:
# Finner optimalt kvantum for følgerbedrift ved å sette q1_sol inn i reaksjonsfunksjonen til bedrift 2
q2_sol = q2.subs(q_1, q1_sol)

display(q2_sol)

6500.00000000000

In [10]:
# Optimal pris i likevekt (sluttmarkedet)
optimal_pris = Invers_etterspørsel.subs({q_1: q1_sol, q_2: q2_sol})

display(optimal_pris)

2450.00000000000

Vi finner at optimalt kvantum for lederbedriften blir 19 000, mens for følgerbedriften blir optimalt kvantum 6 500.

I sluttmarkedet vil markedsprisen bli 2450.

In [11]:
# Profitt for lederbedrift
profitt_1 = (optimal_pris - c_1)*q1_sol-Faste_kostnader

display(sp.simplify(profitt_1))

15050000.0000000

In [12]:
# Profitt for følgerbedrift
profitt_2 = (optimal_pris - c_2)*q2_sol-Faste_kostnader

display(sp.simplify(profitt_2))

1225000.00000000

Profitten til bedriftene blir:

15 050 000 for lederbedriften hurtigruten expeditions og 1 225 000 for følgerbedriften nykommeren.